transaction/std/src/transaction_builder.rs

```diff
@@ -22,12 +22,22 @@ use mc_transaction_core::{
 };
 use mc_util_from_random::FromRandom;
 use rand_core::{CryptoRng, RngCore};
+use std::cmp::Ordering;
 
-/// This is a datatype for the comparer of the transaction outputs
-pub type TxComparerFn = fn(&TxOut, &TxOut) -> core::cmp::Ordering;
+/// A trait used to compare the transaction outputs
+pub trait TxOutputsOrdering {
+    /// comparer method
+    fn cmp(a: &CompressedRistrettoPublic, b: &CompressedRistrettoPublic) -> Ordering;
+}
```

Comment:
A trait seems like maybe too much work to achieve this? Would this be useful elsewhere, if so that could justify this approach.

Reply:
what other way would you recomend?
cc: @jcape

Reply:
This was my suggestion, because I don't think we should be afraid of generics, traits, and static dispatch.

In a normal build, there's no reason to ever change this sorting, so we shouldn't allow access to change it, and with a trait/functor, the compiler should be able to generate identical assembly for this vs. the original.